tm2/pkg/sdk/auth/ante.go

```diff
@@ -406,10 +406,10 @@ func EnsureSufficientMempoolFees(ctx sdk.Context, fee std.Fee) sdk.Result {
 }
 
 // SetGasMeter returns a new context with a gas meter set from a given context.
-func SetGasMeter(simulate bool, ctx sdk.Context, gasLimit int64) sdk.Context {
+func SetGasMeter(ctx sdk.Context, gasLimit int64) sdk.Context {
 	// In various cases such as simulation and during the genesis block, we do not
 	// meter any gas utilization.
-	if simulate || ctx.BlockHeight() == 0 {
+	if ctx.BlockHeight() == 0 {
```

Comment:
Context for this change is: we don't differentiate a regular tx and a simulate tx; they are both valid cases and the same for the gas meter. It saves a lot of headaches if we don't have diverging flows